src/ReplaySession.cc

```diff
@@ -873,10 +873,14 @@ Completion ReplaySession::emulate_async_signal(
          * check again if we're at the
          * target. */
         LOG(debug) << "    trap was for target $ip";
-        /* (The breakpoint would have trapped
-         * at the $ip one byte beyond the
-         * target.) */
-        DEBUG_ASSERT(!at_target);
+        if (t->arch() == x86 || t->arch() == x86_64) {
+          /* (The breakpoint would have trapped
+          * at the $ip one byte beyond the
+          * target.) */
+          DEBUG_ASSERT(!at_target);
+        } else {
+          DEBUG_ASSERT(regs.ip() == t->ip());
```

Comment:
This is just a duplicate of the preceding assertion with undo_executed_bkpt in it.

Reply:
True under the assumption that all non-x86 architectures have the same breakpoint behavior as aarch64. I wanted to defend against the possibility that that might not be the case such that a future architecture implementer could update the list of platforms here properly. Alternatively, we could just get rid of both branches of this if/else, since I don't think the x86 side adds too much either.

Reply:
I'd be fine with removing it entirely.

Reply:
Will do